src/rtk/features/delegableApi/delegableApiSlice.ts

```diff
@@ -47,24 +69,82 @@ const delegableApiSlice = createSlice({
   name: 'delegableApi',
   initialState,
   reducers: {
-    softAdd: (state, action) => {
+    softAdd: (state: SliceState, action: any) => {
       const { delegableApiList } = state;
+      const { presentedApiList } = state;
+      const { delegableApiSearchPool } = state;
       state.delegableApiList = delegableApiList.filter(
         (delegableApi) => delegableApi.id !== action.payload.id,
       );
+      state.presentedApiList = presentedApiList.filter(
+        (delegableApi) => delegableApi.id !== action.payload.id,
+      );
+      state.delegableApiSearchPool = delegableApiSearchPool.filter(
+        (delegableApi) => delegableApi.id !== action.payload.id,
+      );
 
       state.chosenDelegableApiList.push(action.payload);
     },
-    softRemove: (state, action) => {
+    softRemove: (state: SliceState, action: any) => {
       state.delegableApiList.push(action.payload);
+      state.presentedApiList.push(action.payload);
+      state.delegableApiSearchPool.push(action.payload);
 
       const { chosenDelegableApiList } = state;
       state.chosenDelegableApiList = chosenDelegableApiList.filter(
         (delegableApi) => delegableApi.id !== action.payload.id,
       );
     },
+    filter: (state: SliceState, action: any) => {
+      const { delegableApiList } = state;
+      const filterList = action.payload;
+      let searchPool = [...delegableApiList];
+      if (filterList && filterList.length > 0) {
+        searchPool = delegableApiList.filter((api) =>
+          filterList.some((filter: string) => {
+            if (api.orgName.toLocaleLowerCase().includes(filter.toLowerCase())) {
+              return true;
+            }
+            return false;
+          }),
+        );
+        state.delegableApiSearchPool = searchPool;
+      } else {
+        state.delegableApiSearchPool = delegableApiList;
+      }
+    },
+    search: (state: SliceState, action: any) => {
+      const { delegableApiSearchPool } = state;
+      const searchText = action.payload.trim().toLowerCase();
+      const seachWords = searchText ? searchText.split(' ') : [];
+
+      const prioritizedApiList: DelegableApiWithPriority[] = [];
```

Comment:
One edge case here. When I write E.g. A in searchfield it seems like the prioritation gets opposite from the default state. Is there any reason for this?
![image](https://user-images.githubusercontent.com/31471142/206713243-8d497f5c-413d-4892-a420-9e7eb8d059da.png)

Reply:
I guess that's just a quirk of the .sort function since that's the one I use for sorting...